src/application/service/action.go

```diff
@@ -76,14 +76,14 @@ func (self *actionService) Save(tx pgx.Tx, action *domain.Action) error {
 	return nil
 }
 
-func (self *actionService) GetCurrent(tx pgx.Tx) (actions []*domain.Action, err error) {
+func (self *actionService) GetCurrent() (actions []*domain.Action, err error) {
 	self.logger.Debug().Msg("Getting current Actions")
-	actions, err = self.actionRepository.GetCurrent(tx)
+	actions, err = self.actionRepository.GetCurrent()
 	err = errors.WithMessagef(err, "Could not select current Actions")
 	return
 }
 
-func (self *actionService) IsRunnable(tx pgx.Tx, action *domain.Action) (bool, map[string]interface{}, error) {
+func (self *actionService) IsRunnable(action *domain.Action) (bool, map[string]interface{}, error) {
```

Comment:
The transaction is necessary here to avoid a race condition (facts could change in between calls to `getInputFacts()` and `getInputFactLatest()`).

Reply:
Both methods are doing a SELECT on the database. You can pass the Tx as a parameter but it is exactly the same that uses the db to query the information.
We have the quote: "FIXME race condition: facts may change in between checking whether each input is satisfied"
 
One way to resolve the race condition in this case is to lock the TABLE for read mode. Or the mode proposed by Postgresql. Is not elegant, but is an option.

WDYT @dermetfan @manveru

Reply:
> lock the TABLE for read mode

This is what the transaction [does](https://www.postgresql.org/docs/9.4/sql-set-transaction.html):

> A statement can only see rows committed before it began. This is the default.

Manually using a `LOCK` statement instead is probably trickier.

Reply:
**A statement can only see rows committed before it began**. 

Make sense, in this point we only require this condition as solution to iterate the range of action.Inputs. I'll revert this change.